File: eLitServer/source/database_classes/drink_category.py
from typing import Dict

import mongoengine as me
from database_classes import *
from database_classes.drink_object import DrinkObject


class DrinkCategory(DrinkObject):
    name = me.StringField(required=True, unique=True)
    image = me.StringField()

    def __init__(self, name: str, image: str = None, *args, **values):
        super().__init__(*args, **values)
        self.name = name
        self.image = image

    def to_dict(self) -> Dict:
        data = super().to_dict()
        data['name'] = self.name
        data['image'] = self.image
        return data

File: eLit/Model/DrinkParts/DrinkCategory.swift
//
//  DrinkCategory.swift
//  eLit
//
//  Created by Gianpaolo Di Pietro on 14/02/2019.
//  Copyright © 2019 eLit.app. All rights reserved.
//

import UIKit
import CoreData

@objc(DrinkCategory)
class DrinkCategory: DrinkObjectWithImage {
    //MARK: Initializers
    
    convenience init(dict: [String: Any]) {
        
        // TODO CHeck existence
        
        self.init()
        self.id = dict["id"] as? String ?? ""
        self.fingerprint = dict["fingerprint"] as? String ?? ""
        self.name = dict["name"] as? String ?? ""
        self.imageURLString = dict["image"] as? String ?? ""
    }
    
    override func update(with data: [String : Any], savePersistent: Bool) {
        self.name = data["name"] as? String ?? ""
        self.imageURLString = data["image"] as? String ?? ""
        self.image = UIImage()
        self.imageData = nil
        super.update(with: data, savePersistent: savePersistent)
    }

}
